libcloud/http.py

```diff
@@ -102,7 +102,8 @@ def set_http_proxy(self, proxy_url):
         self.http_proxy_used = True
 
         self.session.proxies = {
-            self.proxy_scheme: proxy_url
+            'http': proxy_url,
```

Comment:
I tested it and it works fine with just a single attribute set, but it also shouldn't hurt to set both (http and https).

Reply:
This was definitely broken for me. I tested with various versions of requests just in case. Perhaps you have other mitigating factors, like setting the http_proxy and https_proxy environment variables, or only making http requests and not https, or vice versa? 

Anyway, like you say it can't hurt to set both, and the code's been merged, so yay!

If I have some spare time, I'll looking into writing some integration tests for this.